# LargePD CNV

This repository contains the scripts for Quality Control (QC) steps and CNV calling with [Large PD](https://large-pd.org/) dataset from the study: "Genome-wide Analysis of Copy Number Variation in Latin American Parkinson’s Disease Patients"

**Usage**:
`code`  

    $ script
    
**Software**:

1. PLINK 1.9: Chang CC, Chow CC, Tellier LCAM, Vattikuti S, Purcell SM, Lee JJ (2015) Second-generation PLINK: rising to the challenge of larger and richer datasets. GigaScience, 4. URL: https://www.cog-genomics.org/plink2/

2. ADMIXTURE: Alexander DH, Novembre J, Lange K. (2009) Fast model-based estimation of ancestry in unrelated individuals. Genome Research, 19:1655–1664, 2009. URL: http://software.genetics.ucla.edu/admixture/

3. KING: Manichaikul A, Mychaleckyj JC, Rich SS, Daly K, Sale M, Chen WM (2010) Robust relationship inference in genome-wide association studies. Bioinformatics 26(22):2867-2873. URL: http://people.virginia.edu/~wc9c/KING/

4. PennCNV: Wang K, Li M, Hadley D, Liu R, Glessner J, Grant S, Hakonarson H, Bucan M (2007) An integrated hidden Markov model designed for high-resolution copy number variation detection in whole-genome SNP genotyping data. Genome Res. 17, 1665–1674. URL: http://penncnv.openbioinformatics.org/en/latest/

5. R: R Core Team (2019). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL: https://www.R-project.org/

